### PR TITLE
consumer part

### PR DESCRIPTION
**Pull request by dvasunin**

### CODE REVIEW

File: src/main/java/com/dih/connector/test/client/connector/DataspaceConnectorConfiguration.java [hunk 1]
```diff
@@ -15,4 +36,44 @@
     public BasicAuthRequestInterceptor adminAuth() {
         return new BasicAuthRequestInterceptor(username, password);
     }
+
+    @Bean
+    public BasicAuthenticationInterceptor authInterceptor() {
```
Comment (daniil-pankratov) [inline src/main/java/com/dih/connector/test/client/connector/DataspaceConnectorConfiguration.java:42]: Why do we need one more basic auth interceptor?

File: src/main/java/com/dih/connector/test/service/ConsumeOfferService.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    private final RestTemplate restTemplateLd;
+
+    @Qualifier("json-default")
+    private final RestTemplate restTemplateDefault;
+
+    private ObjectMapper objectMapper;
+
+    @PostConstruct
+    public void init() {
+        objectMapper = new ObjectMapper();
```
Comment (daniil-pankratov) [inline src/main/java/com/dih/connector/test/service/ConsumeOfferService.java:52]: Maybe we can move ObjectMapper bean creation into configuration?

File: src/main/java/com/dih/connector/test/client/connector/DataspaceConnectorConfiguration.java [hunk 3]
```diff
@@ -15,4 +36,44 @@
...
+        return new BasicAuthenticationInterceptor(username, password);
+    }
+
+    @Bean(name = "json-ld")
```
Comment (daniil-pankratov) [inline src/main/java/com/dih/connector/test/client/connector/DataspaceConnectorConfiguration.java:46]: Why do we need to use RestTemplate? We could create separate configuration for feign client only for specific interfaces (like `/api/ids`) where we could put this http message converter

Comment (dvasunin) [reply src/main/java/com/dih/connector/test/client/connector/DataspaceConnectorConfiguration.java:46]: I do not use explicit model here, rather a generic JsonNode as a container for any message. IMHO RestTemplate is more appropriate here (I would use lower level HTTP client even for that, but RestTemplate is ok as well)

File: src/main/java/com/dih/connector/test/service/ConsumeOfferService.java (hunk 2 above)
Comment (dvasunin) [reply src/main/java/com/dih/connector/test/service/ConsumeOfferService.java:52]: it is not a system-wide one, used for debugging purpose only. To pretty print the messages.

File: src/main/java/com/dih/connector/test/client/connector/DataspaceConnectorConfiguration.java (hunk 1 above)
Comment (dvasunin) [reply src/main/java/com/dih/connector/test/client/connector/DataspaceConnectorConfiguration.java:42]: BasicAuthenticationInterceptor != BasicAuthRequestInterceptor

**Review by daniil-pankratov — APPROVED** [review]

**Pull request merged**